go-controller/pkg/cni/cni.go

```diff
@@ -174,9 +174,24 @@ func (pr *PodRequest) cmdAddWithGetCNIResultFunc(kubeAuth *KubeAPIAuth, clientse
 			if err != nil {
 				return nil, err
 			}
-			if _, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo); err != nil {
+			primaryUDNResult, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo)
+			if err != nil {
 				return nil, err
 			}
+
+			defaulClusterNetworkResult := response.Result
+			originalNumberOfIfaces := len(primaryUDNResult.Interfaces)
+			response.Result.Interfaces = append(primaryUDNResult.Interfaces, response.Result.Interfaces...)
+			for i := range response.Result.IPs {
+				ifaceIPConfig := defaulClusterNetworkResult.IPs[i].Copy()
+
+				if ifaceIPConfig.Interface != nil {
+					ifaceIPConfig.Interface = current.Int(*ifaceIPConfig.Interface + originalNumberOfIfaces)
+				}
+				response.Result.IPs = append(primaryUDNResult.IPs, ifaceIPConfig)
```

Comment:
Can we update the CNI test to make the default network dual stack as well? Because I am not sure how would this work in that case.

Reply:
If we always append to `primaryUDNResult.IPs` but save in `response.Result.IPs` we might not end up will all the IPs we want if a new slice backend is allocated in append.

Reply:
Good catch !!!

Reply:
Added dual stack info to the unit test